vllm/__init__.py

```diff
@@ -1,5 +1,20 @@
 """vLLM: a high-throughput and memory-efficient inference engine for LLMs"""
 
+import os
+
+# Set OMP_NUM_THREADS to 1 if it is not already set
+# Helps to avoid CPU contention, particularly when running in a container and
+# when using multiprocessing.
+#
+# This must be set before importing torch to have the full effect. It seems that
+# torch determines the number of threads it will use when it is imported.
+if "OMP_NUM_THREADS" not in os.environ:
```

Comment:
Hi @tjohnson31415 , can you exclude the CPU backend here? I'm afraid setting ```OMP_NUM_THREADS=1``` by default will have some side affects. You can use ```is_cpu()``` in the ```utils.py```, thanks :)

Reply:
Good Point! I wasn't thinking about the CPU backend :)
I just pushed up a change to use `torch.set_num_threads()` instead of needing OMP_NUM_THEADS set before `import torch`, so the code is moved back to the gpu executor.